react/admin/ReturnsTable/ReturnsTableContent.tsx

```diff
@@ -80,88 +77,93 @@ class ReturnsTableContent extends Component<IProps, IState> {
+        this.setState({ isFiltered: false })
+      } else {
+        this.setState({ isFiltered: true })
+      }
 
-    if (filters.orderId !== '') {
-      where += ` AND orderId="*${filters.orderId}*"`
-    }
+      if (filters.orderId !== '') {
+        where += ` AND orderId="*${filters.orderId}*"`
+      }
 
-    if (filters.returnId !== '') {
-      where += ` AND id="*${filters.returnId}*"`
-    }
+      if (filters.returnId !== '') {
+        where += ` AND id="*${filters.returnId}*"`
+      }
 
-    if (filters.sequenceNumber !== '') {
-      where += ` AND sequenceNumber="${filters.sequenceNumber}"`
-    }
+      if (filters.sequenceNumber !== '') {
+        where += ` AND sequenceNumber="${filters.sequenceNumber}"`
+      }
 
-    let startDate = '1970-01-01'
-    let endDate = currentDate()
+      let startDate = '1970-01-01'
+      let endDate = currentDate()
 
-    if (filters.fromDate !== '' || filters.toDate !== '') {
-      startDate =
-        filters.fromDate !== '' ? filterDate(filters.fromDate) : startDate
-      endDate =
-        filters.toDate !== ''
-          ? filterDate(filters.toDate)
-          : filterDate(filters.fromDate)
+      if (filters.fromDate !== '' || filters.toDate !== '') {
+        startDate =
+          filters.fromDate !== '' ? filterDate(filters.fromDate) : startDate
+        endDate =
+          filters.toDate !== ''
+            ? filterDate(filters.toDate)
+            : filterDate(filters.fromDate)
 
-      where += ` AND createdIn between ${startDate} AND ${endDate}`
-    }
+        where += ` AND createdIn between ${startDate} AND ${endDate}`
+      }
 
-    if (filters.status !== '') {
-      where += ` AND status="${requestsStatuses[filters.status]}"`
-    }
+      if (filters.status !== '') {
+        where += ` AND status="${requestsStatuses[filters.status]}"`
+      }
 
-    const url = `${
-      fetchPath.getRequests + schemaNames.request
-    }/${pageNumber}/${perPage}/${sortedBy}/${sortOrder}/${where}`
+      const url = `${
+        fetchPath.getRequests + schemaNames.request
+      }/${pageNumber}/${perPage}/${sortedBy}/${sortOrder}/${where}`
 
-    const returnsResponse = await fetch(url)
+      const returnsResponse = await fetch(url)
 
-    const returns = await returnsResponse.json()
+      const returns = await returnsResponse.json()
 
-    if ('error' in returns) {
-      this.setState({ error: returns.error, tableIsLoading: false })
-    } else {
-      this.setState({
-        itmes: returns.data,
-        // orderedItems: returns,
-        totalItems: returns.pagination?.total,
-        tableIsLoading: false,
-      })
+      if ('error' in returns) {
+        this.setState({ error: returns.error, tableIsLoading: false })
+      } else {
+        this.setState({
+          itmes: returns.data,
+          totalItems: returns.pagination?.total,
+          tableIsLoading: false,
+        })
+      }
+    } catch (error) {
+      this.setState({ error: error.message })
```

Comment:
Here I think we should set state to `tableIsLoading: false`

Reply:
correct! i think there are two places to add `tableIsLoading: false`

Reply:
You can always add a finally block, like [here](https://github.com/vtex-apps/admin-pages/blob/bdf655a3530b528292af89c1ee7f9b4627572c8f/react/components/admin/store/PWAForm/index.tsx#L119-L121)